micrometer-registry-iotdb/src/main/java/org/apache/iotdb/micrometer/IoTDBRegistry.java

```diff
@@ -0,0 +1,225 @@
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.cumulative.CumulativeCounter;
+import io.micrometer.core.instrument.cumulative.CumulativeDistributionSummary;
+import io.micrometer.core.instrument.cumulative.CumulativeFunctionCounter;
+import io.micrometer.core.instrument.cumulative.CumulativeFunctionTimer;
+import io.micrometer.core.instrument.cumulative.CumulativeTimer;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.internal.DefaultGauge;
+import io.micrometer.core.instrument.internal.DefaultLongTaskTimer;
+import io.micrometer.core.instrument.internal.DefaultMeter;
+import io.micrometer.core.instrument.push.PushMeterRegistry;
+import io.micrometer.core.instrument.util.NamedThreadFactory;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.iotdb.rpc.IoTDBConnectionException;
+import org.apache.iotdb.rpc.StatementExecutionException;
+import org.apache.iotdb.session.pool.SessionPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+/**
+ * Registry which stores all values as IoTDB Time Series.
+ * All time-related values should be milliseconds.
+ * <p>
+ * With default settings, each second all values are written.
+ */
+public class IoTDBRegistry extends PushMeterRegistry {
+
+    private static final Logger logger = LoggerFactory.getLogger(IoTDBRegistry.class);
+    private final SessionPool sessionPool;
+
+    public IoTDBRegistry(IoTDBRegistryConfig config, Clock clock) {
+        super(config, clock);
+
+        sessionPool = new SessionPool(config.getIp(), config.getPort(), config.getUser(), config.getPassword(), 1);
+
+        // Prepare a metric for here...
+        start(new NamedThreadFactory("iotdb-metrics-publisher"));
+    }
+
+    @Override
+    protected <T> Gauge newGauge(Meter.Id id, T obj, ToDoubleFunction<T> valueFunction) {
+        return new DefaultGauge<>(id, obj, valueFunction);
+    }
+
+    @Override
+    protected Counter newCounter(Meter.Id id) {
+        return new CumulativeCounter(id);
```

Comment:
the CumulativeCounter uses an array (DoubleAdder) to records each increased value. But the counter has only an interface count() . Is that needed to save all historical increased values (Should we change to some other counter implementation)?